### ISSUE TITLE
0.5.0 Introduces regression when compiling queries

### ISSUE THREAD

**Issue by kdichev** [gatsbyjs/gatsby-graphql-toolkit#11]
after upgrading the compiled queries are quite strange and apply fragments on types that I don't expect to see there

example 0.4.0:

```
query LIST_accordionListItem($limit: Int, $skip: Int) {
  accordionListItemCollection(limit: $limit, skip: $skip) {
    total
    items {
      remoteTypeName: __typename
      ...AccordionListItemId
      ...AccordionListItem
    }
  }
}

fragment AccordionListItemId on AccordionListItem {
  remoteTypeName: __typename
  sys {
    id
  }
}

fragment AccordionListItem on AccordionListItem {
  sys {
    ... on Sys {
      id
      spaceId
      environmentId
      publishedAt
      firstPublishedAt
      publishedVersion
    }
  }
  entryTitle
  title
  description
  content {
    remoteTypeName: __typename
    ... on PaxTimeTable {
      remoteTypeName: __typename
      sys {
        id
      }
    }
    ... on TextAndMediaLayout {
      remoteTypeName: __typename
      sys {
        id
      }
    }
    ... on PageContentTextSection {
      remoteTypeName: __typename
      sys {
        id
      }
    }
  }
}
```

after 0.5.0:
```
query LIST_accordionListItem($limit: Int, $skip: Int) {
  accordionListItemCollection(limit: $limit, skip: $skip) {
    total
    items {
      remoteTypeName: __typename
      ...AccordionListItemId
      ...PageContentAccordionList__itemsCollection__0__items__0
      ...AccordionListItem
      ...SupportFormLogistics__emailTemplate__0
    }
  }
}

fragment AccordionListItemId on AccordionListItem {
  remoteTypeName: __typename
  sys {
    id
  }
}

fragment PageContentAccordionList__itemsCollection__0__items__0 on AccordionListItem {
  remoteTypeName: __typename
  sys {
    remoteTypeName: __typename
    id
  }
}

fragment AccordionListItem on AccordionListItem {
  sys {
    remoteTypeName: __typename
    ... on Sys {
      id
      spaceId
      environmentId
      publishedAt
      firstPublishedAt
      publishedVersion
    }
  }
  entryTitle
  title
  description
  content {
    remoteTypeName: __typename
    ... on PaxTimeTable {
      remoteTypeName: __typename
      sys {
        remoteTypeName: __typename
        id
      }
    }
    ... on TextAndMediaLayout {
      remoteTypeName: __typename
      sys {
        remoteTypeName: __typename
        id
      }
    }
    ... on PageContentTextSection {
      remoteTypeName: __typename
      sys {
        remoteTypeName: __typename
        id
      }
    }
  }
}

fragment SupportFormLogistics__emailTemplate__0 on Entry {
  remoteTypeName: __typename
  sys {
    remoteTypeName: __typename
    id
  }
}
```

**Comment by vladar** [comment]
Yeah, we have a prettify step to get rid of those redundant fragments. But looks like it doesn't work as expected in 0.5.0. In theory, it shouldn't break anything but I'll check it.

Thanks for reporting! 👍

**Comment by kdichev** [comment]
In my case the issue is that I have a query complexity limit from api and those fragments push the query over it

**Comment by vladar** [comment]
I think the problem is with nested ids. The prettify step basically removes duplicates of ID fragment. But in your case after we add `remoteTypeName: __typename` to `sys` field it becomes different from ID fragment:

```graphql
fragment AccordionListItemId on AccordionListItem {
  remoteTypeName: __typename
  sys {
    id
  }
}
```

I think in v0.5.0 this is what've changed:

```diff
fragment PageContentAccordionList__itemsCollection__0__items__0 on AccordionListItem {
  remoteTypeName: __typename
  sys {
+   remoteTypeName: __typename
    id
  }
}
```

Now, this fragment is not a subset of the ID fragment anymore hence not removed. Just out of curiosity - what happens if you add it to the ID fragment (not suggesting as a fix, just want to confirm):

```diff
fragment AccordionListItemId on AccordionListItem {
  remoteTypeName: __typename
  sys {
+   remoteTypeName: __typename
    id
  }
}
```

**Comment by vladar** [comment]
Should be fixed in `0.6.0`
